main/src/tokenizer/nllb_tokenizer.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2022 The Facebook AI Research Team Authors and The HuggingFace Inc. team.
+// Copyright 2019-2021 Guillaume Becquin
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+
+use crate::vocab::EXTENDED_FAIRSEQ_LANGUAGE_CODES;
+use crate::{
+    error::TokenizerError,
+    vocab::{NLLBVocab, SentencePieceBpeModel, Vocab},
+    Mask, Offset, OffsetSize, Token, TokenIdsWithOffsets, TokenIdsWithSpecialTokens,
+};
+
+use super::{
+    tokenization_utils::{clean_text, decompose_nfkc, is_whitespace, split_on_language_code},
+    MultiThreadedTokenizer, Tokenizer,
+};
+
+pub struct NLLBTokenizer {
+    model: SentencePieceBpeModel,
+    vocab: NLLBVocab,
+    src_lang: String,
+}
+
+impl NLLBTokenizer {
+    pub fn from_files<V: AsRef<Path>, M: AsRef<Path>, S: AsRef<Path>>(
```

Comment:
I'm sorry for another small items: all tokenizers have 2 methods:
- `from_file(s)`: creation without special token map (using the defaults)
- `from_file(s)_with_special_token_map`: creation with a special token map

Could the methods please be aligned with the existing tokenizers?

Reply:
Done.